app/process_proposal.go

```diff
@@ -130,24 +128,18 @@ func (app *App) ProcessProposalHandler(ctx sdk.Context, req *abci.RequestProcess
 
 	}
 
-	dataSquare, err := square.Construct(req.Txs, app.MaxEffectiveSquareSize(ctx), appconsts.SubtreeRootThreshold)
+	eds, err := da.ConstructEDS(req.Txs, appconsts.Version, app.MaxEffectiveSquareSize(ctx))
 	if err != nil {
-		logInvalidPropBlockError(app.Logger(), blockHeader, "failure to compute data square from transactions:", err)
+		logInvalidPropBlockError(app.Logger(), blockHeader, "failure to compute extended data square from transactions:", err)
 		return reject(), nil
 	}
 
 	// Assert that the square size stated by the proposer is correct
-	if uint64(dataSquare.Size()) != req.SquareSize {
+	if uint64(eds.Width()) != req.SquareSize*2 {
 		logInvalidPropBlock(app.Logger(), blockHeader, "proposed square size differs from calculated square size")
 		return reject(), nil
 	}
 
-	eds, err := da.ExtendShares(share.ToBytes(dataSquare))
```

Comment:
What was the point of this check before ? Just to ensure that constructed square can be decomposed back into shares, and then extended again into square? But there's no equality check

Reply:
Are you talking about `ExtendShares`? That was part of constructing the DAH